jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/NodeInstanceImpl.java

```diff
@@ -149,9 +155,13 @@ public Node getNode() {
     public boolean isInversionOfControl() {
         return false;
     }
+
+    public final void cancel() {
+        cancel(ABORTED);
```

Comment:
Field aborted and cancelType aren't necessary to be persisted long-term, but more short-term for data passing.  Probably aborted could have been transient as well, but for backwards compability we probably can't change that now.  So I guess the plan is to just keep the field, nothing more.  I think we have tests that are run as part of QE suite that test portability of process instance binary state across versions, can we make sure these are run to make sure we don't break backwards compatibility with this?

Reply:
Yes, there are tests as part of a test suite in `jbpm-runtime-manager` module. However these are related to a migration between different bpm processes. So I'm wondering whether this will be enough or we should add a specific test case for backward compatibility. @krisv, @elguardian wdyt?

https://github.com/kiegroup/jbpm/blob/master/jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/migration/MigrationManagerTest.java

Reply:
This aborted field would be persisted with any node instance, so I think any of those migration tests would be able to detect migration issues caused by this.  There should be sufficient other unit tests for simply testing whether aborting a node instance works well (using cancelType).